lib/matplotlib/tests/test_widgets.py

```diff
@@ -1757,3 +1757,21 @@ def test_MultiCursor(horizOn, vertOn):
         assert l.get_xdata() == (.5, .5)
     for l in multi.hlines:
         assert l.get_ydata() == (.25, .25)
+
+
+def test_parent_axes_removal():
+
+    fig, (ax_radio, ax_checks) = plt.subplots(1, 2)
+
+    radio = widgets.RadioButtons(ax_radio, ['1', '2'], 0)
+    checks = widgets.CheckButtons(ax_checks, ['1', '2'], [True, False])
+
+    ax_checks.remove()
+    ax_radio.remove()
+    with io.BytesIO() as out:
+        fig.savefig(out, format='raw')
+
+    renderer = fig._get_renderer()
+    evt = DrawEvent('draw_event', fig.canvas, renderer)
+    radio._clear(evt)
+    checks._clear(evt)
```

Comment:
This is obviously some kind of smoke test as it does not contain assertions. A short description would be helpful though.

I understand that removal of the buttons should not brick the figure. But what exactly are we testing with the savefig, and what with the _clear calls? (I assume there are actually two separate tests?